compiler/rustc_target/src/spec/mod.rs

```diff
@@ -520,6 +535,92 @@ impl ToJson for LinkerFlavorCli {
     }
 }
 
+/// The different `-Clink-self-contained` options that can be specified in a target spec:
+/// - enabling or disabling in bulk
+/// - some target-specific pieces of inference to determine whether to use self-contained linking
+///   if `-Clink-self-contained` is not specified explicitly (e.g. on musl/mingw)
+/// - explicitly enabling some of the self-contained linking components, e.g. the linker component
+///   to use `rust-lld`
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum LinkSelfContained {
```

Comment:
```suggestion
pub enum LinkSelfContainedDefault {
```
This looks like `LinkSelfContainedDefault`.

I see that the old `LinkSelfContainedDefault` is still used for some compatibility code, but it seems better to remove it right away.
Or rather add `WithComponents` to it instead of introducing a new enum.